ec2api/api/security_group.py

```diff
@@ -229,6 +231,17 @@ def _validate_parameters(protocol, from_port, to_port):
         if (to_port < -1 or to_port > 255) and (to_port != 65535):
             raise exception.InvalidParameterValue(
                 _('ICMP code is out of range.'))
+    elif protocol in ['any']:
+        if from_port > to_port:
+            raise exception.InvalidParameterValue(
+                _('Invalid port range.'))
+        if from_port < -1 or from_port > 65535:
+            raise exception.InvalidParameterValue(
+                _('TCP/UDP from port is out of range.'))
```

Comment:
Incorrect exception message since protocol is not TCP/UDP. Same for next exception too